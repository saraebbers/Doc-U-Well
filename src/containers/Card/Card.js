import React, { Component } from 'react';
import '../../index.scss';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Card extends Component {

  render() {

    switch (this.props.type) {
      case 'profile' :
        const { dob, height, weight, bp_systolic, bp_diastolic, heart_rate, blood_type } = this.props.attributes
        const { id } = this.props
        let currentProfile = this.props.profile.find(file => {
          return file.id == id
        })
        const profileName = `${currentProfile.attributes.given_name} ${currentProfile.attributes.surname}`.toUpperCase()
        const dobOptions = {
          month: 'long',
          day: 'numeric',
          year: 'numeric',
        }
        const birthDate = new Date(dob).toLocaleString('en-US', dobOptions)

        return(
          <div className='card-container profiletype'>
            <h3> { profileName }</h3>
            <h4> DOB: { birthDate }</h4>
            <p>Blood type: { blood_type } </p>
            <p>Height: { height } inches </p>
            <p>Weight: { weight } lbs</p>
            <p>Blood Pressure: { bp_systolic } / { bp_diastolic } </p>
            <p>Heart Rate: { heart_rate } bpm</p>
          </div>
        )
      case 'appointments' :
        const { datetime, provider_id, notes} = this.props.attributes
        let currentProvider = this.props.providers.find(provider => {
          return provider.id == provider_id
        })
        const options = {
          weekday: 'long',
          month: 'long',
          day: 'numeric',
          hour: '2-digit',
          minute: '2-digit',
          year: 'numeric',
        }
        const date = new Date(datetime * 1000).toLocaleString('en-US', options);
        const providerName = `${ currentProvider.attributes.given_name } ${ currentProvider.attributes.surname }`.toUpperCase()

        return(
          <div className='card-container appointmentstype'>
            <i className="fas fa-trash-alt"></i>
            <h3> APPOINTMENT WITH</h3>
            <h3> {providerName} </h3>
            <h4> {date} </h4>
            <p> { currentProvider.attributes.street_address } { currentProvider.attributes.city } { currentProvider.attributes.state } { currentProvider.attributes.zip }</p>
            <p> { currentProvider.attributes.phone } </p>
            <p> { notes } </p>

          </div>
        )
      case 'providers' :
        const { speciality, given_name, surname, street_address, city, state, zip, phone  } = this.props.attributes
        let specialityType

        if(speciality) {
         specialityType = speciality.toUpperCase()
       } else {
         specialityType = ''
       }
        
        return(
            <div className='card-container providerstype'>
            <h3> {specialityType} PROVIDER </h3>
            <i className="fas fa-trash-alt"></i>
            <h4> { given_name } { surname }</h4>
            <p>{ street_address } { city } { state } { zip }</p>
            <p>{ phone } </p>
          </div>
        )
      case 'insurance' :
        const { insurance_type, carrier, id_number, group_number, phone_number } = this.props.attributes
        let insuranceType

        if(insurance_type) {
          insuranceType = insurance_type.toUpperCase()
        } else {
          insuranceType = 'not entered'
        }
        return(
          <div className='card-container insurancetype'>
            <i className="fas fa-trash-alt"></i>
            <h3> { insuranceType } CARD</h3>
            <h4> { carrier } Policy Number: { id_number } </h4>
            <p> Group: { group_number } </p>
            <p> { carrier } Phone Number: { phone_number } </p>
          </div>
        )
      case 'other' :
        const { message } = this.props
        return (
          <div className='card-container othertype'>
            <p> { message} </p>
          </div>
          )
      default:
        return ('hit card default')
    }
  }
}

// Card.propTypes = {
//   name: PropTypes.string
// }

export const mapStateToProps = (state) => {
  return {
    appointments: state.appointments,
    providers: state.providers,
    insurance: state.insurance,
    profile: state.profile,
    isLoading: state.isLoading,
    errorMessage: state.errorMessage,
    }
}

export default connect(mapStateToProps)(Card);
