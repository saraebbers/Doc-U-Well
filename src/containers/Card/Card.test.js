import React from 'react';
import { Card, mapStateToProps } from './Card';
import { shallow } from 'enzyme';

describe('Card Container', () => {
  let mockprofile
  let mockappointments
  let mockproviders
  let mockinsurance
  let mockother
  let mockattributes
  let mockmessage
  let mockCurrentProvider
  let mockprovider
  let mockprofile123

beforeEach(()=> {

  mockprofile = "profile"
  mockappointments = "appointments"
  mockproviders = "providers"
  mockinsurance = "insurance"
  mockother = "other"
  mockattributes = { speciality: 'dental', given_name: 'Bob', surname: 'Smith', street_address: '123', city: 'London', state: 'ND', zip: '12345', phone: '1234567890', dob: '12121212', height: '65', weight: '123', bp_systolic: '120', bp_diastolic: '67', heart_rate: '22', blood_type: 'A+', datetime: '12121212', provider_id: '1', notes: 'tf7yg8ouhp' }
  mockmessage = "this is a message"
  mockprofile123 = [{provider: {id: '1', attributes: {given_name: 'Sally', sirname: 'Jones'}}, attributes: {given_name: 'Robert'}}]
})

  it('should match the screenshot', () => {
      const wrapper = shallow(<Card />)
      expect(wrapper).toMatchSnapshot()
    })

  it('should render the correct information if passed providers as props', () => {
    const wrapper = shallow(<Card type={mockproviders} attributes={mockattributes}/>)
    expect(wrapper.find('.providerstype').length).toEqual(1);
  })

  it('should render the correct information if passed profile as props', () => {
    const wrapper = shallow(<Card type={mockprofile} attributes={mockattributes} profile={mockprofile123}/>)
    expect(wrapper.find('.profiletype').length).toEqual(1);
  })


  it('should render the correct information if passed insurance as props', () => {
    const wrapper = shallow(<Card type={mockinsurance} attributes={mockinsurance}/>)
    expect(wrapper.find('.insurancetype').length).toEqual(1);
  })

  it('should render the correct information if passed other as props', () => {
    const wrapper = shallow(<Card type={mockother} message={mockmessage}/>)
    expect(wrapper.find('.othertype').length).toEqual(1);
  })

})
