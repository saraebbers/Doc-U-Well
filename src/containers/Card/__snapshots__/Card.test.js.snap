// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "hit card default",
  Symbol(enzyme.__nodes__): Array [
    "hit card default",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
